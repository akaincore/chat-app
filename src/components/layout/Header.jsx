import AppBar from '@material-ui/core/AppBar/index';
import React from 'react';
import Toolbar from '@material-ui/core/es/Toolbar/Toolbar';
import Typography from '@material-ui/core/es/Typography/Typography';
import { withStyles } from '@material-ui/core/styles';


const styles = () => ({
  appBar: {
    width: '100%',
  },
});

function Header(props) {
  const { classes } = props;

  return (
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <Typography variant="h6" color="inherit">
          Chat App
        </Typography>
      </Toolbar>
    </AppBar>
  );
}

export default withStyles(styles)(Header);
