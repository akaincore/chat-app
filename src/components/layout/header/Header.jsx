import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar/index';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import ProfileMenu from './ProfileMenu';
import ChatMenu from './ChatMenu';
import Chat from '../../../types/Chat';
import CurrentUser from '../../../types/CurrentUser';

const styles = () => ({
  appBar: {
    width: '100%',
  },
  toolBar: {
    padding: '0 50px',
    display: 'flex',
    justifyContent: 'space-between',
  },
  titleWrapper: {
    display: 'flex',
    alignItems: 'center',
  },
});

const Header = ({
  classes,
  logout,
  isAuthenticated,
  currentUser,
  updateUser,
  activeChat,
  deleteChat,
  leaveChat,
  isConnected,
}) => (
  <AppBar position="static" className={classes.appBar}>
    <Toolbar className={classes.toolBar}>
      <div className={classes.titleWrapper}>
        <Typography variant="h6" color="inherit">
          {activeChat ? activeChat.title : 'Chat App'}
        </Typography>
        {activeChat && (
          <ChatMenu
            disabled={!isConnected}
            currentUser={currentUser}
            deleteChat={() => deleteChat(activeChat._id)}
            leaveChat={() => leaveChat(activeChat._id)}
          />
        )}
      </div>
      {isAuthenticated && (
        <ProfileMenu
          disabled={!isConnected}
          currentUser={currentUser}
          updateUser={updateUser}
          logout={logout}
        />
      )}
    </Toolbar>
  </AppBar>
);

Header.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  logout: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  currentUser: PropTypes.shape(CurrentUser).isRequired,
  updateUser: PropTypes.func.isRequired,
  activeChat: PropTypes.shape(Chat),
  deleteChat: PropTypes.func.isRequired,
  leaveChat: PropTypes.func.isRequired,
  isConnected: PropTypes.bool.isRequired,
};

Header.defaultProps = {
  activeChat: null,
};

export default withStyles(styles)(Header);
