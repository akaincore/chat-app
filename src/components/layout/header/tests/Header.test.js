/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../Header';

jest.mock('../ChatMenu', () => () => 'ChatMenu');
jest.mock('../ProfileMenu', () => () => 'ProfileMenu');

const mockProps = {
  currentUser: {
    username: 'user',
    isMember: false,
    isChatMember: false,
    isCreator: false,
  },
  logout: jest.fn(),
  updateUser: jest.fn(),
  deleteChat: jest.fn(),
  leaveChat: jest.fn(),
  isAuthenticated: false,
  isConnected: false,
  activeChat: {
    _id: '12345',
    creator: { username: 'joe' },
    title: 'foo',
  },
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header {...mockProps} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
