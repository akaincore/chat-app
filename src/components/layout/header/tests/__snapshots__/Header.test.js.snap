// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header
  className="MuiPaper-root-92 MuiPaper-elevation4-98 MuiAppBar-root-83 MuiAppBar-positionStatic-87 MuiAppBar-colorPrimary-90 Header-appBar-80"
>
  <div
    className="MuiToolbar-root-119 MuiToolbar-regular-121 MuiToolbar-gutters-120 Header-toolBar-81"
  >
    <div
      className="Header-titleWrapper-82"
    >
      <h6
        className="MuiTypography-root-123 MuiTypography-h6-140 MuiTypography-colorInherit-152"
      >
        foo
      </h6>
    </div>
  </div>
</header>
`;
